app/utils/date_on_or_after_validator.rb

```diff
@@ -0,0 +1,19 @@
+class DateOnOrAfterValidator < ActiveModel::Validator
```

Comment:
This guy feels like a bit too much hard coded assumptions to me. Could it e.g. be a more general validation that validates some given date field is on or after another given datefield? Then the record fields used would be parameterized instead of assuming start_on and end_on.